packages/plugin-core/src/components/views/PreviewPanel.ts

```diff
@@ -314,23 +307,66 @@ export class PreviewPanel implements PreviewProxy, vscode.Disposable {
       previous.contentHash !== current.contentHash,
   });
 
-  private sendRefreshMessage(panel: vscode.WebviewPanel, note: NoteProps) {
-    const syncChangedNote = true;
-    note = this.rewriteImageUrls(note, panel);
+  /**
+   * Notify preview webview panel to display latest contents
+   *
+   * @param panel panel to notify
+   * @param note note to display
+   * @param isFullRefresh If true, sync contents of note with what's being seen in active editor.
+   * This will be true in cases where user switches between tabs or opens/closes notes without saving, as contents of notes may not match engine notes.
+   * Otherwise display contents of note
+   */
+  private async sendRefreshMessage(
+    panel: vscode.WebviewPanel,
+    note: NoteProps,
+    isFullRefresh: boolean
+  ) {
+    if (this.isVisible()) {
+      // Engine state has not changed so do not sync. This is for displaying updated text only
+      const syncChangedNote = false;
+
+      // If full refresh is required, sync note with contents in active text editor
+      const textDocument = VSCodeUtils.getActiveTextEditor()?.document;
+      if (textDocument && isFullRefresh) {
+        note = await this._textDocumentService.applyTextDocument(
+          note,
+          textDocument
+        );
+      }
+      note = this.rewriteImageUrls(note, panel);
 
-    return panel.webview.postMessage({
-      type: DMessageEnum.ON_DID_CHANGE_ACTIVE_TEXT_EDITOR,
-      data: {
-        note,
-        syncChangedNote,
-      },
-      source: "vscode",
-    } as OnDidChangeActiveTextEditorMsg);
+      // This is necessary so that the cache will display these contents since it checks `updated` property
+      const now = NoteUtils.genUpdateTime();
```

Comment:
Btw - one way you can tell if the cached version is being used is that you can open up the server.log during debugging under `packages/plugin-core/dendron.server.log` - when you open up preview / switch tabs, the logs will indicate if it used a cached version or not; you can also see render times.

Reply:
It looks like we only depend on `updated` prop for determining cache freshness: https://github.com/dendronhq/dendron/blob/5930f4aa81bda22b6ab0e53817c3ed965b1127a1/packages/engine-server/src/enginev2.ts#L614

It might make more sense to use `contentHash` instead.

With current implementation, I'll need to update each time, even on editor tab changes. For example:
1. Type something in note A
2. Close note A without saving. Goes to note B
3. Open note A again. Preview will show note A after step 1 was done, when I want to see it before step 1

Reply:
I think it makes sense to use `contentHash`. That's also what I went with when caching `rewriteImageUrls`, right above this function.

Reply:
I'm going to add another prop to store `contentHash` in the cache. I also think we need to add another check (probably in separate pr) to check if backlinks have also changed as I don't believe those affect `contentHash`